src/System.Management.Automation/FormatAndOutput/common/BaseOutputtingCommand.cs

```diff
@@ -743,6 +743,27 @@ private void ProcessCachedGroupOnWide(WideViewHeaderInfo wvhi, List<PacketInfoDa
             _formattingHint = hint;
         }
 
+        static private int GetConsoleWindowWidth(int columnNumber)
+        {
+            if (columnNumber == int.MaxValue)
+            {
+                // Tables and Wides need to use spaces for padding to maintain table look even if console window is resized.
+                // For all other output, we use int.MaxValue if the user didn't explicitly specify a width.
+                // If we detect that int.MaxValue is used, first we try to get the current console window width.
+                // However, if we can't read that (for example, implicit remoting has no console window), we default
+                // to something reasonable: 120 columns.
+                try
+                {
+                    return Console.WindowWidth;
+                }
+                catch
+                {
+                    return 120;
```

Comment:
I think we can safely assert that if trying to get the WindowWidth ever fails, it will always fail.  So if we catch here, I think we can set a private flag like _failedToReadConsoleWidth and just return the default rather than incurring the cost of handling the exception each time (I think that was @iSazonov's original concern)

Reply:
Does it really always fail?

I will present a simple example.   
I simply add `_failedToReadConsoleWidth` flag as follows.

```csharp
static private int _failedToReadConsoleWidth = 0;

static private int GetConsoleWindowWidth(int columnNumber)
{
    if (columnNumber == int.MaxValue)
    {
        // Tables and Wides need to use spaces for padding to maintain table look even if console window is resized.
        // For all other output, we use int.MaxValue if the user didn't explicitly specify a width.
        // If we detect that int.MaxValue is used, first we try to get the current console window width.
        // However, if we can't read that (for example, implicit remoting has no console window), we default
        // to something reasonable: 120 columns.
        if (_failedToReadConsoleWidth > 0)
        {
            return _failedToReadConsoleWidth;
        }
        try
        {
            return Console.WindowWidth;
        }
        catch
        {
            _failedToReadConsoleWidth = 120;
            return _failedToReadConsoleWidth;
        }
    }
    return columnNumber;
}
```

then invoke four commands in order.

* command1 : Invoke local command (Console.WindowWidth is available)
* command2 : Invoke implict remoting command (Console.WindowWidth isn't available)
* command3 : Invoke implict remoting command (Console.WindowWidth isn't available)
* command4 : Invoke local command (Console.WindowWidth is available)

value of `_failedToReadConsoleWidth` is changed to 120 after command2, and this value is used in command3.
But when we invoke command4, I think we have to use `Console.WindowWidth`.

Reply:
We should reset the flag.

Reply:
Hmm...

In order to reset the flag, we need to be able to determine if `Console.WindowWidth` can get.  
if we can determine whether we can get `Console.WindowWidth` from the start, we do not need to use `try-catch` statement.

I am sorry but I have no idea that how to determine if `Console.WindowWidth` can get...

Reply:
In the implicit remoting case, the command would be executed on the remote side where it wouldn't be able to read the WindowWidth.  So the cached value would only be the remote side.  Local commands should still work.

Reply:
@SteveL-MSFT Thank you. My premise was wrong.
I will fix the code as above.